### PR TITLE
[NON-MODULAR] Feline Trait

### PR DESCRIPTION
**Pull request by linnpap**
## About The Pull Request
Adds a new quirk that gives people feline traits, such as being attracted to lasers, and also meowing when they speak.

## Why It's Good For The Game
im FUCKING TIRED of HAVING TO ROLEPLAY TAY BEING ATTRACTED TO LASERS its FUCKING BULLSHIT BRO HUBERT CAN JUST AUTOMATICALLY STARE AT LASERS BUT NO, I HAVE TO FUCKING MANUALLY DRAG MY FUCKING MOUSE, AND CLICK IN THE GENERAL DIRECTION OF THE LASER. I PLAY AS A FUCKING CAT. THIS IS BULLSHIT.

## Changelog
:cl:
add: Added new 'Feline Traits' quirk
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
Time to pick the feline trait as a felinid for epic trolling

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -114,3 +114,12 @@
 				H.emote("laugh")
 				addtimer(CALLBACK(H, /mob/proc/emote, "laugh"), 5 SECONDS)
 				addtimer(CALLBACK(H, /mob/proc/emote, "laugh"), 10 SECONDS)
+
+/datum/quirk/feline_aspect
+	name = "Feline Traits"
+	desc = "You happen to act like a feline, for whatever reason."
+	gain_text = "<span class='notice'>Nya~ could go for some catnip wight about now...~ OwO</span>"
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm]: That's just really ugly, find something less "uwu" please.

Comment (linnpap) [reply modular_skyrat/master_files/code/datums/traits/neutral.dm]: angy-bladerunner

**Comment by KasparoVy** [comment]
Damn I'm down for this, cute as hell. Thanks much!

**Comment by SpaceVampire** [comment]
Aren't we in the middle of a feature freeze?
https://github.com/Skyrat-SS13/Skyrat-tg/pull/7367

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 2]
```diff
@@ -114,3 +114,12 @@
 				H.emote("laugh")
 				addtimer(CALLBACK(H, /mob/proc/emote, "laugh"), 5 SECONDS)
 				addtimer(CALLBACK(H, /mob/proc/emote, "laugh"), 10 SECONDS)
+
+/datum/quirk/feline_aspect
+	name = "Feline Traits"
+	desc = "You happen to act like a feline, for whatever reason."
```
Comment (axietheaxolotl) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm:120]:
```suggestion
	desc = "You happen to act like a felinid, for whatever reason."
```

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 3]
```diff
@@ -114,3 +114,12 @@
...
+	gain_text = "<span class='notice'>Nya could go for some catnip right about now...</span>"
+	lose_text = "<span class='notice'>You feel less attracted to lasers.</span>"
+	medical_record_text = "Patient seems to possess behavior much like a feline."
```
Comment (axietheaxolotl) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm:123]:
```suggestion
	medical_record_text = "Patient seems to possess behavior much like a felinid."
```

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 4]
```diff
@@ -114,3 +114,12 @@
 				H.emote("laugh")
 				addtimer(CALLBACK(H, /mob/proc/emote, "laugh"), 5 SECONDS)
 				addtimer(CALLBACK(H, /mob/proc/emote, "laugh"), 10 SECONDS)
+
+/datum/quirk/feline_aspect
+	name = "Feline Traits"
```
Comment (axietheaxolotl) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm:119]:
```suggestion
	name = "Felinid Traits"
```
dumbass

**Review by axietheaxolotl — CHANGES_REQUESTED** [review]
It's felinid not feline, you idiotic cat.

**Comment by linnpap** [comment]
> 
> It's felinid not feline, you idiotic cat.

im gonna beat the shit out of you

**Comment by axietheaxolotl** [comment]
Yeah, in your dreams.

**Comment by axietheaxolotl** [comment]
> 
> 
> > It's felinid not feline, you idiotic cat.
> 
> im gonna beat the shit out of you

yeah, in your dreams

File: modular_skyrat/master_files/code/datums/traits/neutral.dm (hunk 4 above)
Comment (thestubborn) [reply modular_skyrat/master_files/code/datums/traits/neutral.dm:119]: felind is the neko race

feline is a name to refer to the members of the felidae family

File: code/game/objects/items/devices/laserpointer.dm [hunk 5]
```diff
@@ -135,7 +135,8 @@
 
 	//catpeople
 	for(var/mob/living/carbon/human/H in view(1,targloc))
-		if(!isfelinid(H) || H.incapacitated() || H.is_blind())
+		//if(!isfelinid(H) || H.incapacitated() || H.is_blind())
+		if(!(isfelinid(H) || HAS_TRAIT(H, TRAIT_FELINE)) || H.incapacitated() || H.is_blind()) //SKYRAT EDIT - FELINE TRAIT
```
Comment (tf-4) [inline code/game/objects/items/devices/laserpointer.dm:139]:
```suggestion
		if(!isfelinid(H) || !HAS_TRAIT(H, TRAIT_FELINE) || H.incapacitated() || H.is_blind()) //SKYRAT EDIT - FELINE TRAIT
```

I feel like the way you wrote it is kinda unclear. Maybe just me though.

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 6]
```diff
@@ -114,3 +114,12 @@
...
+	desc = "You happen to act like a feline, for whatever reason."
+	gain_text = "<span class='notice'>Nya could go for some catnip right about now...</span>"
```
Comment (tf-4) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm]:
```suggestion
	gain_text = span_notice("Nya could go for some catnip right about now...")
```

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 7]
```diff
@@ -114,3 +114,12 @@
...
+	lose_text = "<span class='notice'>You feel less attracted to lasers.</span>"
```
Comment (tf-4) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm]:
```suggestion
	lose_text = span_notice("You feel less attracted to lasers.")
```

File: code/game/objects/items/devices/laserpointer.dm (hunk 5 above)
Comment (KasparoVy) [reply code/game/objects/items/devices/laserpointer.dm:139]: This is clearer at a glance, don't see no reason why not to use it :D
You even save a character.

Comment (linnpap) [reply code/game/objects/items/devices/laserpointer.dm:139]: without the parenthesizes, it means that you have to be a felinid AND have the trait to be affected, meaning it completely breaks it
i made sure to test this too, and sure enough, it doesnt work without them

**Pull request merged**